firebase-firestore/src/androidTest/java/com/google/firebase/firestore/ValidationTest.java

```diff
@@ -755,6 +763,72 @@ public void queriesUsingInAndDocumentIdMustHaveProperDocumentReferencesInArray()
 
   // Helpers
 
+  /** Performs a write using each write API and makes sure it succeeds. */
+  private static void expectWriteSuccess(Object data) {
+    expectWriteSuccess(data, /*includeSets=*/ true, /*includeUpdates=*/ true);
+  }
+
+  /** Performs a write using each update API and makes sure it succeeds. */
+  private static void expectUpdateSuccess(Map<String, Object> data) {
+    expectWriteSuccess(data, /*includeSets=*/ false, /*includeUpdates=*/ true);
+  }
+
+  /** Performs a write using each set API and makes sure it succeeds. */
+  private static void expectSetSuccess(Object data) {
+    expectWriteSuccess(data, /*includeSets=*/ true, /*includeUpdates=*/ false);
+  }
```

Comment:
Are you intentionally including expectUpdateSuccess() and expectSetSuccess() even though they're not used?  If so, that's fine...  just double-checking.

Reply:
I added the equivalent facility for checking success that exists for checking failure on the premise that if it were available we'd use it.